README.rst

```diff
@@ -59,6 +59,77 @@ Next, let's parse something and get an abstract syntax tree:
 You'd typically then use a ``nodes.NodeVisitor`` subclass (see below) to walk
 the tree and do something useful with it.
 
+Another example would be to implement a parser for `.ini`-files. Consider the following:
+
+.. code:: python
+
+    grammar = Grammar(
+        r"""
+        expr        = (entry / emptyline)*
+        entry       = section (ws pair)*
+        section     = lpar word rpar
+        key         = word+
+        value       = (word / quoted)+
+        pair        = key equal value
+        word        = ~r"[-\w]+"
+        quoted      = ~'"[^\"]+"'
+        equal       = ws? "=" ws?
+        lpar        = "["
+        rpar        = "]"
+        ws          = ~"\s*"
+        emptyline   = ws+
+        """
+    )
+
+We could now implement a subclass of `NodeVisitor` like so:
+
+.. code:: python
+
+    class IniVisitor(NodeVisitor):
+        output = {}
+        current_section = None
+        current_key = None
+
+        def generic_visit(self, node, visited_children):
+            return node.text or visited_children
+
+        def visit_section(self, node, visited_children):
+            self.current_section = node.text[1:-1]
+            self.output[self.current_section] = {}
+
+        def visit_key(self, node, visited_children):
+            self.current_key = node.text
+
+        def visit_value(self, node, visited_children):
+            self.output[self.current_section][self.current_key] = node.text
+            
+And call it like that:
+
+.. code:: python
+
+    from parsimonious.grammar import Grammar
+    from parsimonious.nodes import NodeVisitor
+
+    data = """[section]
+    somekey = somevalue
+    someotherkey=someothervalue
+
+    [anothersection]
+    key123 = "what the heck?"
+    key456="yet another one here"
+
+    """
+
+    tree = grammar.parse(data)
+    iv = IniVisitor()
+    iv.visit(tree)
+    print(iv.output)
```

Comment:
It's a little unusual to squirrel away the output in a class-level var. It means you can't use 2 instances of the class at once, as the data leaks between them:

```
>>> class Foo(object):
...     out = []
... 
>>> f = Foo()
>>> g = Foo()
>>> f.out.append(5)
>>> g.out
[5]
```

Rather, we should return the output from each visitation method, culminating in the final output squirting out of `visit()`.

Reply:
Do you have an example? I had difficulties with the hierarchy here.

Reply:
You mean we need to overwrite the `visit` def in each subclass as well?

Reply:
What I'm saying is that each visit_* method returns something, which is then passed as an arg to the visit_* method that corresponds to the surrounding construct. For example, if you are parsing the expression `a * (a + b)`, visit_parenthetical might return its result to visit_multiplication. Here are a couple of complete code examples:

* This one turns parsed Python-ish regexes into JavaScript ones: https://github.com/mozilla/dxr/blob/master/dxr/trigrammer.py#L430
* This one turns queries like "someFilter:argument someOtherFilter:anotherArgument" into a dict summary: https://github.com/mozilla/dxr/blob/master/dxr/query.py#L271